### PR TITLE
[url_launcher] Adds lint ignores to the usage of `RouteInformation.location`

### PR DESCRIPTION
**Pull request by chunhtai**
flutter framework is about to deprecate RouteInformation.location. Preparing this plugin for soft-transition

Relevant issue: https://github.com/flutter/flutter/issues/100624
Relevant PR: https://github.com/flutter/flutter/pull/119968

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [ ] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
-## NEXT
+## 2.1.2
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md:1]: The CHANGELOG and version change should be reverted per https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates (this is a dev-only change, and thus exempt from both).

File: packages/url_launcher/url_launcher_platform_interface/test/link_test.dart [hunk 2]
```diff
@@ -83,6 +83,9 @@ class _RouteDelegate extends RouterDelegate<RouteInformation>
     if (_history.isEmpty) {
       return const Placeholder(key: Key('empty'));
     }
+    // TODO(chunhtai): remove the ignore once migration completes.
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_platform_interface/test/link_test.dart:86]: Could you summarize in the comment what "once migration completes" means concretely? The issue you linked to is really long, so it's not immediately obvious.

File: packages/url_launcher/url_launcher_platform_interface/test/link_test.dart [hunk 3]
```diff
@@ -83,6 +83,9 @@ class _RouteDelegate extends RouterDelegate<RouteInformation>
...
+    // https://github.com/flutter/flutter/issues/100624.
+    // ignore: unnecessary_string_interpolations, deprecated_member_use
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_platform_interface/test/link_test.dart:88]: Why is unnecessary_string_interpolations here?

**Review by stuartmorgan-g — CHANGES_REQUESTED** [review]

File: packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (chunhtai) [reply packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md:1]: I think i need to publish a new version for linux plugin_test to pass? https://github.com/flutter/flutter/pull/119968/checks?check_run_id=11180655690

File: packages/url_launcher/url_launcher_platform_interface/test/link_test.dart (hunk 3 above)
Comment (chunhtai) [reply packages/url_launcher/url_launcher_platform_interface/test/link_test.dart:88]: The change also makes location non nullable

File: packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/url_launcher/url_launcher_platform_interface/CHANGELOG.md:1]: Nope, that test is running on a pinned+rolled copy of this repository.

**Comment by stuartmorgan-g** [comment]
We've just completed the migration of the plugin code to the flutter/packages repository, as described in https://flutter.dev/go/flutter-plugins-repo-migration, and this repository is now being archived. Unfortunately that means that all in-progress PRs here must be moved to flutter/packages.

Please see [our instructions](https://github.com/flutter/flutter/wiki/Migrating-Plugins-repository-PRs-to-Packages) for an explanation of how to move your PR, and if you have any issues moving your PR please don't hesitate to reach out in the #hackers-ecosystem channel in [Discord](https://github.com/flutter/flutter/wiki/Chat).

Our apologies that your PR was caught in this one-time transition. We're aware that it's disruptive in the short term, and appreciate your help in getting us to a better long-term state!
